src/libs/actions/PersonalDetails.js

```diff
@@ -123,11 +132,15 @@ function fetch() {
             const allPersonalDetails = formatPersonalDetails(data.personalDetailsList);
             Onyx.merge(ONYXKEYS.PERSONAL_DETAILS, allPersonalDetails);
 
-            const myPersonalDetails = allPersonalDetails[currentUserEmail]
-                || {avatarURL: getAvatar(undefined, currentUserEmail)};
+            const currentUserPersonalDetails = allPersonalDetails[currentUserEmail]
+                || {avatar: getAvatar(undefined, currentUserEmail)};
+
+            // Add the first and last name to the current user's MY_PERSONAL_DETAILS key
+            currentUserPersonalDetails.firstName = data.personalDetailsList[currentUserEmail].firstName ?? '';
+            currentUserPersonalDetails.lastName = data.personalDetailsList[currentUserEmail].lastName ?? '';
 
             // Set my personal details so they can be easily accessed and subscribed to on their own key
-            Onyx.merge(ONYXKEYS.MY_PERSONAL_DETAILS, myPersonalDetails);
+            Onyx.merge(ONYXKEYS.MY_PERSONAL_DETAILS, currentUserPersonalDetails);
```

Comment:
NAB switching this to `currentUserPersonalDetails` seems unnecessary because `myPersonalDetails` is the key we are storing.